### ISSUE TITLE
Add build expression example to README before this tool lands in nixpkgs

### ISSUE THREAD

**Issue by paluh** [misuzu/nix-simple-deploy#1]
Thanks for building this tool!

I'm a newbie user of NixOS. It has taken me more than a few minutes to build rust package expression (I'm not even sure if I've done this in the most idiomatic way) for my `nix-shell` to work with your nice tool.
I'm not sure what is the best practice in such a case but from my perspective it would be cool to add a simple example how to do this to README before `nix-simple-deploy` lands in nixpkgs. What do you think?

**Comment by misuzu** [comment]
Done. Not sure about `nix-shell`, haven't used it that way.

**Comment by paluh** [comment]
Wow! That was really quick - thanks!

I wonder if maybe we can turn `nix-simple-deploy.nix` into `default.nix` which `src` equals to `./.` (I've seen this approach in other packages from NUR like here: https://github.com/Mic92/nixos-shell/blob/master/default.nix) so it will be a bit easier to fetch and build given version of `nix-simple-deploy`.
I've experimented with this and it works nicely: https://github.com/paluh/nix-simple-deploy/blob/master/default.nix.

It would be a matter of rev and hash to get final expression then and we don't have to hardcode any hash into provided expression:

``` nix
let
  nix-simple-deploy = pkgs.callPackage (pkgs.fetchFromGitHub {
    rev = "...";
      owner = "misuzu";
      repo = "nix-simple-deploy";
      sha256 = "...";
  }) {};
in
# ... add it to buildInputs in case of `nix-shell` or to `environment.systemPackages` in case of system configuration
```

Should I provide a PR?

**Comment by misuzu** [comment]
> Should I provide a PR?

Sure, go ahead.

**Comment by paluh** [comment]
Closed by #2
